crates/cli/src/modules/webhooks.rs

```
@@ -0,0 +1,60 @@
+use crate::help_strings::HelpStrings;
+use clap::{Args, Subcommand};
+use std::str;
+use tungstenite::{connect, Message};
+use url::Url;
+use serde_json::{from_str, to_string_pretty, Value};
+use colored::Colorize;
+
+
+/// Webhooks options and flags
+#[derive(Args)]
+#[clap(about = HelpStrings::Schema)]
+pub struct WebhooksOptions {
+    /// All the subcommands of the Webhooks cli
+    #[clap(subcommand)]
+    pub commands: WebhooksSubcommands,
+}
+
+/// Webhooks subcommands
+#[derive(Subcommand, Debug)]
+#[clap(about = HelpStrings::Webhooks)]
+pub enum WebhooksSubcommands {
+    /// Listen for webhooks on provided url
+    #[clap(about = HelpStrings::Webhooks)]
+    Listen {},
+}
+
+/// Listen to webhooks for an agentURL
+pub fn listen(agent_url: String) -> ! {
+    // TODO: filter by/listen to by topic
+    let stripped_agent_url = match &agent_url {
+        s if s.starts_with("http://") => &s[7..], 
+        s if s.starts_with("https://") => &s[8..],
+        s => s,
```

This is not correct and will panic below.

Is neither of the cases are met we should return an error and bubble it up.

---

hmm I see. Was thinking that there might be a url without prefix in there... But I guess you do have a point here - better to assume the url is wrong altogether